### PR TITLE
CVSL-1786 Moving to use the new approval endpoints

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: server/@types/licenceApiClientTypes.ts [hunk 1]
```diff
@@ -15,7 +15,11 @@ export type AdditionalConditionData = components['schemas']['AdditionalCondition
 export type ContactNumberRequest = components['schemas']['ContactNumberRequest']
 export type CreateLicenceRequest = components['schemas']['CreateLicenceRequest']
 export type LicenceSummary = components['schemas']['LicenceSummary']
-export type Licence = components['schemas']['Licence']
+export type Licence =
+  | components['schemas']['VariationLicence']
+  | components['schemas']['HardStopLicence']
+  | components['schemas']['CrdLicence']
```
Comment (andrewrlee) [inline server/@types/licenceApiClientTypes.ts:21]: Had to manually delete a union from the generated licencesApiClientTypes and manually implement here to get round a bug where Licence and CrdLicence were treated as `any`.

This is hopefully due to a bug which will be fixed by [this](https://github.com/drwpow/openapi-typescript/pull/1578)

File: server/routes/approvingLicences/handlers/approvalCases.ts [hunk 2]
```diff
@@ -20,7 +20,9 @@ export default class ApprovalCaseRoutes {
     const allPrisons = await this.prisonerService.getPrisons()
     const activeCaseload = allPrisons.filter(p => p.agencyId === user.activeCaseload)
     const prisonCaseloadToDisplay = caseloadsSelected.length ? caseloadsSelected : [activeCaseload[0].agencyId]
-    const cases = await this.caseloadService.getApproverCaseload(user, prisonCaseloadToDisplay, approvalNeededView)
+    const cases = approvalNeededView
+      ? await this.approverCaseloadService.getApprovalNeeded(user, prisonCaseloadToDisplay)
+      : await this.approverCaseloadService.getRecentlyApproved(user, prisonCaseloadToDisplay)
```
Comment (andrewrlee) [inline server/routes/approvingLicences/handlers/approvalCases.ts:25]: now calling new service

File: server/services/approverCaseloadService.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  async getRecentlyApproved(user: User, prisonCaseload: string[]): Promise<ManagedCaseForApproval[]> {
+    const licences: LicenceSummaryApproverView[] = await this.licenceApiClient.getLicencesRecentlyApproved(
+      filterCentralCaseload(prisonCaseload),
+      user
+    )
+    const caseLoad: ManagedCaseForApproval[] = await this.mapLicencesToOffenders(licences)
+    return this.mapResponsibleComsToCases(caseLoad)
+  }
+
+  private pairDeliusRecordsWithNomis = async (
```
Comment (andrewrlee) [inline server/services/approverCaseloadService.ts:37]: bit of copy paste from the caseload service. 

I think this is reasonable as a relatively short term fix - it feels like it's unlikely that there will be a change required that will affect all caseloads at the same time and its probably easier to reimplement this in the backend in isolation from the other caseloads

Comment (andrewrlee) [reply server/services/approverCaseloadService.ts:37]: I held off refactoring this further - there's some gnarly stuff in here

File: server/services/licenceService.ts [hunk 4]
```diff
@@ -274,26 +274,6 @@ export default class LicenceService {
     return _.head(licences)
   }
 
-  async getLicencesForApproval(user: User, prisonCaseload: string[]): Promise<LicenceSummary[]> {
```
Comment (andrewrlee) [inline server/services/licenceService.ts:277]: no need for these calls to exist here so inlined in the approval caseload service

**Review by nishanthmahasamudram — APPROVED** [review]
LGTM

**Pull request merged**